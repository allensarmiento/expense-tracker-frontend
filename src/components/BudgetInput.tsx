import React, { useState } from 'react';
import Form from 'react-bootstrap/Form';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import styled from 'styled-components';

const BudgetForm = styled(Form)`
  padding: 1rem 2rem;
`;

const DateInput = styled(Form.Control)``;

const DescriptionInput = styled(Form.Control)``;

const AmountInput = styled(Form.Control)``;

const AddButton = styled(Button)`
  width: 100%;
`;

type BudgetInputProps = {
  onAddItem: Function;
};

const BudgetInput = ({ onAddItem }: BudgetInputProps) => {
  const currentDate = new Date();
  const year = currentDate.getUTCFullYear();
  const month = currentDate.getUTCMonth() + 1 < 10
    ? `0${currentDate.getUTCMonth() + 1}`
    : currentDate.getUTCMonth() + 1;
  const day = currentDate.getUTCDate() < 10
    ? `0${currentDate.getUTCDate()}`
    : currentDate.getUTCDate();
  const formattedDate = `${year}-${month}-${day}`;

  const [date, setDate] = useState(formattedDate);
  const [type, setType] = useState('expense');
  const [category, setCategory] = useState('food');
  const [description, setDescription] = useState('');
  const [amount, setAmount] = useState('');

  const addItem = () => {
    if (date && type && category && description && amount) {
      onAddItem({
        date,
        type,
        category,
        description,
        amount: Number.parseFloat(amount) || 0.00
      });

      setDescription('');
      setAmount('');
    }
  };

  return (
    <BudgetForm
      onSubmit={(e: React.FormEvent<HTMLInputElement>) => e.preventDefault()}
    >
      <Form.Row>
        <Col>
          <DateInput
            type="date"
            name="current-date"
            min="1870-01-01"
            max={formattedDate}
            value={date}
            onChange={(e: React.FormEvent<HTMLInputElement>) => {
              setDate(e.currentTarget.value);
            }}
          />
        </Col>
        <Col>
          <Form.Control
            as="select"
            value={type}
            onChange={e => setType(e.target.value)}
          >
            <option value="income">(+) Income</option>
            <option value="expense">(-) Expense</option>
          </Form.Control>
        </Col>
        <Col>
          <Form.Control
            as="select"
            value={category}
            onChange={e => setCategory(e.target.value)}
          >
            <option value="rent">Rent</option>
            <option value="utilities">Utilities</option>
            <option value="food">Food</option>
            <option value="drinks">Drinks</option>
          </Form.Control>
        </Col>
        <Col>
          <DescriptionInput
            placeholder="Description"
            value={description}
            onChange={(e: React.FormEvent<HTMLInputElement>) => {
              setDescription(e.currentTarget.value);
            }}
          />
        </Col>
        <Col>
          <AmountInput
            type="number"
            min="0"
            step=".01"
            placeholder="0.00"
            value={amount}
            onChange={(e: React.FormEvent<HTMLInputElement>) => (
              setAmount(e.currentTarget.value)
            )}
          />
        </Col>
        <Col>
          <AddButton variant="primary" onClick={addItem}>Add</AddButton>
        </Col>
      </Form.Row>
    </BudgetForm>
  );
};

export default BudgetInput;
