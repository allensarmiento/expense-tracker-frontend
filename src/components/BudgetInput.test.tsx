import { shallow } from 'enzyme';
import BudgetInput from './BudgetInput';

const onAddItem = jest.fn();

it('renders without crashing', () => {
  expect(shallow(
    <BudgetInput onAddItem={onAddItem} />
  )).toMatchSnapshot();

  // Test for input and sumbit clicked
});
