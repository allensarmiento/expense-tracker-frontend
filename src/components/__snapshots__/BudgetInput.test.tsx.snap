// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(Form)
  onSubmit={[Function]}
>
  <FormRow>
    <Col>
      <Styled(FormControl)
        max="2021-02-23"
        min="1870-01-01"
        name="current-date"
        onChange={[Function]}
        type="date"
        value="2021-02-23"
      />
    </Col>
    <Col>
      <FormControl
        as="select"
        onChange={[Function]}
        value="expense"
      >
        <option
          value="income"
        >
          (+) Income
        </option>
        <option
          value="expense"
        >
          (-) Expense
        </option>
      </FormControl>
    </Col>
    <Col>
      <FormControl
        as="select"
        onChange={[Function]}
        value="food"
      >
        <option
          value="rent"
        >
          Rent
        </option>
        <option
          value="utilities"
        >
          Utilities
        </option>
        <option
          value="food"
        >
          Food
        </option>
        <option
          value="drinks"
        >
          Drinks
        </option>
      </FormControl>
    </Col>
    <Col>
      <Styled(FormControl)
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
    </Col>
    <Col>
      <Styled(FormControl)
        min="0"
        onChange={[Function]}
        placeholder="0.00"
        step=".01"
        type="number"
        value=""
      />
    </Col>
    <Col>
      <Styled(Button)
        onClick={[Function]}
        variant="primary"
      >
        Add
      </Styled(Button)>
    </Col>
  </FormRow>
</Styled(Form)>
`;
